packages/contracts-bedrock/contracts/L1/L2OutputOracle.sol

```diff
@@ -134,71 +149,58 @@ contract L2OutputOracle is Ownable {
             // blockhash value, and delay submission until it is confident that the L1 block is
             // finalized.
             require(
-                blockhash(_l1Blocknumber) == _l1Blockhash,
-                "Blockhash does not match the hash at the expected height."
+                blockhash(_l1BlockNumber) == _l1Blockhash,
+                "OutputOracle: Blockhash does not match the hash at the expected height."
             );
         }
 
-        l2Outputs[_l2timestamp] = OutputProposal(_l2Output, block.timestamp);
-        latestBlockTimestamp = _l2timestamp;
+        l2Outputs[_l2BlockNumber] = OutputProposal(_l2Output, block.timestamp);
+        latestBlockNumber = _l2BlockNumber;
 
-        emit l2OutputAppended(_l2Output, block.timestamp, _l2timestamp);
+        emit l2OutputAppended(_l2Output, block.timestamp, _l2BlockNumber);
     }
 
     /**
      * @notice Deletes the most recent output.
+     *
      * @param _proposal Represents the output proposal to delete
      */
-    function deleteL2Output(OutputProposal memory _proposal) external onlyOwner {
-        OutputProposal memory outputToDelete = l2Outputs[latestBlockTimestamp];
+    function deleteL2Output(OutputProposal memory _proposal) external onlySequencer {
```

Comment:
We still have an issue here to deal with, which is partially [discussed here](https://github.com/ethereum-optimism/optimism/pull/2598#discussion_r892978171). 

But in summary I see these problems: 
1. It should not be possible to delete outputs back to the full finalization window, as it would allow the sequencer to post a faulty proposal which is immediately finalized. However the FINALIZATION_WINDOW value
2. Even if we could easily add `require(block.timestamp > proposal.timestamp + FINALIZATION_WINDOW)`, 
  2.1. it would still allow the sequencer to delete all outputs in a block, then create a malicious withdrawal in the next.
  2.2. every time a proposal is deleted and replaced, the finalization time gets pushed back to the current `block.timestamp` + the `FINALIZATION_WINDOW`. 
  
The consequences of 2.2 are similar to that of a sequencer which goes offline and is late in posting new outputs, except that it bakes in a method for intentionally delaying withdrawals.